homeassistant/helpers/config_entry_oauth2_flow.py

```diff
@@ -257,6 +259,11 @@ async def async_step_creation(
     ) -> Dict[str, Any]:
         """Create config entry from external data."""
         token = await self.flow_impl.async_resolve_external_data(self.external_data)
+        # Force int for non-compliant oauth2 providers
+        try:
+            token["expires_in"] = int(token["expires_in"])
+        except ValueError as err:
+            _LOGGER.warning("Error converting expires_in to int: %s", err)
```

Comment:
Make sure you abort here too. And add a test for both cases.

Reply:
You didn't abort, you return the token. You need to abort the flow as it's invalid data

Reply:
I'm guessing you mean line #87? How would I go about aborting here?

Reply:
Oh right oops. Sorry.